src/OrchardCore/OrchardCore.Apis.GraphQL.Abstractions/Queries/WhereInputObjectGraphType.cs

```diff
@@ -94,35 +87,78 @@ public virtual void AddScalarFilterFields(Type graphType, string fieldName, stri
 
     private void AddEqualityFilters(Type graphType, string fieldName, string description)
     {
-        AddFilterFields(graphType, EqualityOperators, fieldName, description);
+        AddFilterFields(graphType, CreateGraphType(graphType), EqualityOperators, fieldName, description);
     }
 
     private void AddStringFilters(Type graphType, string fieldName, string description)
     {
-        AddFilterFields(graphType, StringComparisonOperators, fieldName, description);
+        AddFilterFields(graphType, CreateGraphType(graphType), StringComparisonOperators, fieldName, description);
     }
 
     private void AddNonStringFilters(Type graphType, string fieldName, string description)
     {
-        AddFilterFields(graphType, NonStringValueComparisonOperators, fieldName, description);
+        AddFilterFields(graphType, CreateGraphType(graphType), NonStringValueComparisonOperators, fieldName, description);
     }
 
     private void AddMultiValueFilters(Type graphType, string fieldName, string description)
     {
         var wrappedType = typeof(ListGraphType<>).MakeGenericType(graphType);
-        AddFilterFields(wrappedType, MultiValueComparisonOperators, fieldName, description);
+        AddFilterFields(wrappedType, CreateGraphType(graphType), MultiValueComparisonOperators, fieldName, description);
     }
 
-    private void AddFilterFields(Type graphType, IDictionary<string, string> filters, string fieldName, string description)
+    private void AddFilterFields(
+        Type graphType,
+        IGraphType resolvedType,
+        IDictionary<string, string> filters,
+        string fieldName,
+        string description)
     {
         foreach (var filter in filters)
         {
             AddField(new FieldType
             {
                 Name = fieldName + filter.Key,
                 Description = $"{description} {filter.Value}",
-                Type = graphType
+                Type = graphType,
+                ResolvedType = resolvedType,
             });
         }
     }
+
+    private readonly Dictionary<Type, IGraphType> graphTypes = new();
+
+    private IGraphType CreateGraphType(Type type)
```

Comment:
Please double check if we absolutely need to instantiate all of the types here as well. It actually depends on when we are adding those fields.

Reply:
Yes. This is part of the transition from v7 to v8. There are Schema validation that happens, and because we create the dynamic field in the Initialize method instead of the constructor, we get errors like this 

> The field 'Amount' of an Input Object type 'PricePartWhereInput' must have non-null 'ResolvedType' property for all types in the chain.

You may want to try to not set the ResolvedType and see the error.